### ISSUE TITLE
Add support for initial state when using a mapped reducer with modules

### ISSUE THREAD

**Issue by thomhos** [Revisolution/typed-redux-kit#2]
If we want to use a mapped reducer with module configuration we need to pass a default state when initialising the reducer.

Originally we would do this:
```js
const reducer = (state = defaultState, action) ...
```

We could provide it as an argument when making the MappedReducer and set 'this.initialState' in the constructor of the MappedReducer. Then we can add the same logic as above to the .reduce method of the class.

```ts
public reduce = (state: STATE = this.initialState, action: ACTION): STATE => {
    if (!this.reducerMap.has(action.type)) return state
    return this.reducerMap.get(action.type)(state, action)
}
```

How do you guys think about that?

**Comment by Rokt33r** [comment]
Definitely. This should make it work.
```ts
interface Options<S = any> {
  initialState: S
}
class MappedReducer<S> {
  constructor(opts: Options<S>) {
    this.initialState = opts.initialState
  }
}
```

I'll fix it today!

**Comment by thomhos** [comment]
Need to make sure the initial state is optional though in case we don't want to use as module. Cheers!

**Comment by thomhos** [comment]
Fixed in pr #3
